services/src/main/java/org/keycloak/services/resources/admin/UsersResource.java

```diff
@@ -796,7 +796,11 @@ public Response executeActionsEmail(@PathParam("id") String id,
             String link = builder.build(realm.getName()).toString();
             long expiration = TimeUnit.SECONDS.toMinutes(realm.getAccessCodeLifespanUserAction());
 
-            this.session.getProvider(EmailTemplateProvider.class).setRealm(realm).setUser(user).sendExecuteActions(link, expiration);
+            this.session.getProvider(EmailTemplateProvider.class) //
+                        .setRealm(realm) //
+                        .setUser(user) //
+                        .setAttribute("clientId", clientSession.getClient().getClientId()) //
```

Comment:
Is this the proper way to convey the "current" clientId to the templates? I wasn't sure whether it would be okay to introduce a dedicated clientId property on the `EmailTemplateProvider` / `FreemarkerEmailTemplateProvider`